### PR TITLE
session, sessionctx: add unit test for sessionctx, session

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

from 79.8955%  to  80.0779%

### What is changed and how it works?

add unit test and remove unused code.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - N/A

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - N/A

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/10763)

### CODE REVIEW

File: session/txn.go [hunk 1]
```diff
@@ -374,8 +374,6 @@ func mergeToDirtyDB(dirtyDB *executor.DirtyDB, op dirtyTableOperation) {
 		dt.AddRow(op.handle, op.row)
 	case table.DirtyTableDeleteRow:
 		dt.DeleteRow(op.handle)
-	case table.DirtyTableTruncate:
```
Comment (jackysp) [inline session/txn.go]: No used?

File: session/tidb.go [hunk 2]
```diff
@@ -254,6 +232,27 @@ func runStmt(ctx context.Context, sctx sessionctx.Context, s sqlexec.Statement)
 	return rs, err
 }
 
+func checkStmtLimit(ctx context.Context, sctx sessionctx.Context, se *session, sessVars *variable.SessionVars) error {
```
Comment (jackysp) [inline session/tidb.go]: just move the code?

File: session/tidb_test.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ package session
 import (
 	"context"
 	"fmt"
+	"github.com/pingcap/tidb/planner/core"
```
Comment (jackysp) [inline session/tidb_test.go]: move to 3rd libs

File: session/txn.go (hunk 1 above)
Comment (lysu) [reply session/txn.go]: yes

**Pull request merged**